pkg/idms/idms.go

```diff
@@ -0,0 +1,263 @@
+package idms
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	configv1 "github.com/openshift/api/config/v1"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// IDMSBuilder provides struct for configmap object containing connection to the cluster and the configmap definitions.
+type IDMSBuilder struct {
```

Comment:
The name of the package is idms no need to duplicate it in strcut name. Please re-name it to Builder.